lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -104,14 +121,21 @@ def convert_username(value: str) -> str:
 
     Raises:
         ConversionError: input can't be converted
+        UnVerifiedUser: If the username provided does not match with the username encoded the authorization header
 
     Returns:
         internal username representation
     """
     try:
-        return Auth.validate_user(value)
-    except Exception:
-        raise ConversionError(value, "username", type(value).__name__)
+        user, verified = Auth().verify_user(value)
+        if verified or user.is_admin():
+            return value
+        else:
+            raise UnVerifiedUser(value)
+    except Exception as exc:
+        raise ConversionError(
+            value, "username", type(value).__name__
+        ) if not isinstance(exc, UnVerifiedUser) else UnVerifiedUser(value)
```

Comment:
Why not just do:
```
    try:
        user, verified = Auth().verify_user(value)
        if verified or user.is_admin():
            return value
        else:
            raise UnVerifiedUser(value)
    except UnVerifiedUser:
        raise
    except Exception as exc:
        raise ConversionError(
            value, "username", type(value).__name__
        )
```

or even

```
    try:
        user, verified = Auth().verify_user(value)
    except Exception as exc:
        raise ConversionError(
            value, "username", type(value).__name__
        )
    if verified or user.is_admin():
        return value
    else:
        raise UnVerifiedUser(value)
```

Reply:
@webbnh, can you extract this as a discussion topic?  I'd like to expose this thinking more broadly than just a code review comment.

Reply:
Done:  #2277